2011-08-04  Jakub Jelinek  <jakub@redhat.com>

	* gthr-posix.h (__gthread_active_p): Do not use preprocessor
	conditionals and comments inside macro arguments.

--- gcc/gthr-posix.h.jj	2011-07-18 13:17:56.000000000 +0200
+++ gcc/gthr-posix.h	2011-08-04 11:22:40.000000000 +0200
@@ -244,16 +244,15 @@ __gthread_active_p (void)
 static inline int
 __gthread_active_p (void)
 {
-  static void *const __gthread_active_ptr
-    = __extension__ (void *) &__gthrw_(
 /* Android's C library does not provide pthread_cancel, check for
    `pthread_create' instead.  */
 #ifndef __BIONIC__
-				       pthread_cancel
+  static void *const __gthread_active_ptr
+    = __extension__ (void *) &__gthrw_(pthread_cancel);
 #else
-				       pthread_create
+  static void *const __gthread_active_ptr
+    = __extension__ (void *) &__gthrw_(pthread_create);
 #endif
-				       );
   return __gthread_active_ptr != 0;
 }
 
